File: Herencia/python/Microbus.py
from Vehiculo import *

class Microbus(Vehiculo):
    def __init__(self):
        Vehiculo.__init__(self)
        self.largo = 3
        self.asientos = 'Tela de modatelas'
        self.timbre = False

    def sonar(self):
        print("La cucaracha, la cucaracha")

    def gritar(self):
        print("Subale, hay lugares")

    def tocarTimbre(self):
        self.timbre = not self.timbre

File: Herencia/cpp/Microbus.cpp
#include "Microbus.h"
#include "Vehiculo.h"

Microbus::Microbus(){
    largo = 3.0;
    timbre = false;
    asientos = "Tela de modatelas";
}

void Microbus::sonar(){
    cout<<"La cucaracha, la cucaracha"<<endl;
}

void Microbus::gritar(){
    cout<<"Subale, hay lugares"<<endl;
}

void Microbus::tocarTimbre(){
    timbre = !timbre;
}